proto/public/api.proto

```diff
@@ -54,6 +54,50 @@ message TapRequest {
   string path = 11;
 }
 
+// A tap request over kubernetes resources.
+message K8sTapRequest {
+  // Describes the kubernetes pods that should be tapped.
+  ResourceSelection target = 1;
+
+  // Selects over events to be reported.
+  Match match = 2;
+
+  // Limits the number of events to be inspected.
+  float maxRps = 3;
+
+  message Match {
+    oneof match {
+      // If empty, matches all messages.
+      Seq all = 1;
+
+      // If empty, matches no messages.
+      Seq any = 2;
+
+      // Inverts the inner match.
+      Match not = 3;
+
+      // Matches events being sent to any of the selected destinations.
+      ResourceSelection destinations = 4;
```

Comment:
This is quite powerful. Something to keep in mind down the road: we want to avoid making it too easy to enable full fidelity tap across every proxy on your cluster. Sane defaults, and perhaps a guardrail in the CLI indicating how many proxies are about to be tapped, could help mitigate this.

Reply:
and, of course, the controller doesn't have to actually send tap requests to all matching destinations.  In the short term, I don't think we should worry about this too much.